app/Libraries/BeatmapsetMainRuleset.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+declare(strict_types=1);
+
+namespace App\Libraries;
```

Comment:
oh yeah, this and the other beatmapset related library classes should probably go in `App\Libraries\Beatmapset`?